commit 867c7c0451f2e67e715da5b55b5bf6696747fdc3
Author: vishwa <vis.pypi@gmail.com>
Date:   Sun Jun 7 10:16:27 2026 -0400

    Fixed #37130 -- Skipped DB cache deletion when culling offset is zero.

diff --git a/django/core/cache/backends/db.py b/django/core/cache/backends/db.py
index 9663127669..3ede514b70 100644
--- a/django/core/cache/backends/db.py
+++ b/django/core/cache/backends/db.py
@@ -278,8 +278,8 @@ class DatabaseCache(BaseDatabaseCache):
             )
             deleted_count = cursor.rowcount
             remaining_num = num - deleted_count
-            if remaining_num > self._max_entries:
-                cull_num = remaining_num // self._cull_frequency
+            cull_num = remaining_num // self._cull_frequency
+            if cull_num > 0 and remaining_num > self._max_entries:
                 cursor.execute(
                     connection.ops.cache_key_culling_sql() % table, [cull_num]
                 )
